src/Microsoft.Data.Analysis/DataFrame.Arrow.cs

```diff
@@ -29,101 +137,7 @@ public static DataFrame FromArrowRecordBatch(RecordBatch recordBatch)
             foreach (IArrowArray arrowArray in arrowArrays)
             {
                 Field field = arrowSchema.GetFieldByIndex(fieldIndex);
-                IArrowType fieldType = field.DataType;
```

Comment:
All of this is just refactored into the `AppendDataFrameColumnFromArrowArray` method